src/nupic/engine/RegionImplFactory.cpp

```diff
@@ -102,9 +102,9 @@ namespace nupic
 
 
 #if defined(NTA_OS_WINDOWS)
-      const char * filename = "cpp_region.dll";
+      const char * filename = "py_algorithms.dll";
 #else
-      const char * filename = "libcpp_region.so";
+      const char * filename = "libpy_algorithms.so";
```

Comment:
Sorry, I am completely unfamiliar with this module, so I will have to ask some potentially dumb questions. Overall, I think that someone else with more knowledge of the current state and future direction of this code base should complete this code review.

1. If we're going to load the library with a changed name, shouldn't there also be a parallel change in the build of this library that gives the library the new name?
2. What should give us the confidence that renaming the library won't break anything?
3. The name `[lib]py_algorithms` may also be confusing, since we already have `nupic.algorithms`. Also, it may be an unfounded leap of faith that the library will contain something significant besides region code in the future. Someone else with knowledge in the direction of that library will have to chime in here.